### PR TITLE
Update to 3.21.8

### PR DESCRIPTION
**Pull request by cjmartian**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by cjmartian** [comment]
@conda-forge-admin, please rerender

**Comment by dbast** [comment]
The same error all over different CI runs:
```
+ python test_bdist_conda_setup.py bdist_conda --help
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/conda-build_1643129892425/test_tmp/test_bdist_conda_setup.py", line 2, in <module>
    import distutils.command.bdist_conda
ModuleNotFoundError: No module named 'distutils.command.bdist_conda'
Tests failed for conda-build-3.21.8-py39hf3d152e_0.tar.bz2 - moving package to /home/conda/feedstock_root/build_artifacts/broken
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 {% set name = "conda-build" %}
-{% set version = "3.21.7" %}
+{% set version = "3.21.8" %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
   url: https://github.com/conda/conda-build/releases/download/{{ version }}/conda-build-{{ version }}.tar.gz
```
Comment (dbast) [inline recipe/meta.yaml:9]: Can we try:
```suggestion
  url:  https://github.com/conda/conda-build/archive/refs/tags/{{ version }}.tar.gz
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-  sha256: fddb305a72adfb2c618b2c670844949978c36c54ce65c2de6ad836fd0e7e2f87
+  sha256: 0031aff4a34e66340f90428051d480c6c456c89e267f78cf5987126bee68ca5b
```
Comment (dbast) [inline recipe/meta.yaml:10]:
```suggestion
  sha256: 20fb08f117e07d0e489fbe83603c9acbeecdc2544e8d514e122b9cf3d716a6a9
```

Comment (dbast) [reply recipe/meta.yaml:10]: If that works, then the rever generated release tarball is broken.

**Comment by dbast** [comment]
ok. the old version fails with the same error see here https://github.com/conda-forge/conda-build-feedstock/pull/173

**Comment by cjmartian** [comment]
Pinning to `setuptools <60.0.0` seems to have done it.

**Comment by dbast** [comment]
setuptools >=60.0.0 now defaults to their local version of distutils, see https://setuptools.pypa.io/en/latest/history.html#v60-0-0

Setting `SETUPTOOLS_USE_DISTUTILS=stdlib` makes the bdist_conda test work again, as tested here https://github.com/conda-forge/conda-build-feedstock/pull/175

While pinning down the version to `setuptools <60.0.0` works for now, we should figure out how to get https://github.com/conda/conda-build/blob/master/bdist_conda.py working with setuptools >=60.0.0 in the long run.

**Comment by dbast** [comment]
Issue to track this https://github.com/conda/conda-build/issues/4366

File: recipe/meta.yaml [hunk 3]
```diff
@@ -45,7 +45,7 @@ requirements:
     - py-lief
     - pyyaml
     - ripgrep
-    - setuptools
+    - setuptools <60.0.0
```
Comment (dbast) [inline recipe/meta.yaml:48]: Can we add a comment here:
```suggestion
    # setuptools >=60.0.0 brings its own local distutils, which bdist_conda does not work with
    # see also https://setuptools.pypa.io/en/latest/history.html#v60-0-0
    - setuptools <60.0.0
```

Comment (dbast) [reply recipe/meta.yaml:48]: Or even adopt the solution in https://github.com/conda-forge/conda-build-feedstock/pull/175

Comment (cjmartian) [reply recipe/meta.yaml]: Added the build steps from #175

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
+{% set version = "3.21.8" %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
-  url: https://github.com/conda/conda-build/releases/download/{{ version }}/conda-build-{{ version }}.tar.gz
-  sha256: fddb305a72adfb2c618b2c670844949978c36c54ce65c2de6ad836fd0e7e2f87
+  url:  https://github.com/conda/conda-build/releases/download/{{ version }}/conda-build-{{ version }}.tar.gz
```
Comment (dbast) [inline recipe/meta.yaml:9]:
```suggestion
  url: https://github.com/conda/conda-build/releases/download/{{ version }}/conda-build-{{ version }}.tar.gz
```

**Comment by dbast** [comment]
I confirm that I am fine being listed as maintainer.

**Comment by dbast** [comment]
@conda-forge/conda-build Ready for review / merge. Thanks!

**Pull request merged**